### DISCUSSION TITLE
Force download/install a particular bottle for a different platform than host system?

### DISCUSSION THREAD

**Discussion by jbarlow83** [Homebrew/discussions#1601] [Tap maintenance and Homebrew development]
Is there a probably way to ask brew to install a particular bottle

For example, from a macOS 10.15 machine, I'd like to download and install the bottle for arm64_big_sur so that the files can be used in cross compilation. Is there a way to do this short of scraping the formula?

**Comment by carlocab** [comment]
Try this:
```bash
ghp_url () {
        local formula=$1
        local version=$2
        command brew info --json=v1 $formula | jq --join-output ".[0].bottle.stable.files.${version}.url"
}

curl -L -H 'Authorization: Bearer QQ==' $(ghp_url hello arm64_big_sur) | tar -x
```
This will untar the _unrelocated_ bottle tarball, so you'll probably need to make lots of use of `install_name_tool`.

Or you can maybe just download the tarball, make sure it's saved with the correct filename (using `curl`'s `-O` flag _won't_ work), then use `brew install /path/to/tarball`. You'll probably still need to make lots of use of `install_name_tool` anyway, because the ARM bottles are built with a different prefix.

This isn't a supported use-case, by the way, so we can't make promises about this working (and that this will keep working).

**Reply by jbarlow83** [reply]
Thanks for the thorough explanation - sounds like I'm better off doing a direct compile.
